config/config_test.go

```diff
@@ -319,8 +347,20 @@ func TestConfig_Validate(t *testing.T) {
 			false,
 		}, {
 			"valid long",
-			longConfig.Copy(),
+			valid.Copy(),
 			true,
+		}, {
+			"multi-task valid",
+			validMutiTask.Copy(),
+			true,
+		}, {
+			"autocommitting provider reuse error",
+			autoCommit.Copy(),
+			false,
+		}, {
+			"no provider for task error",
+			noProvider.Copy(),
+			false,
```

Comment:
From my understanding, sometimes it is ok for the user to configure a provider block e.g. when the provider can work with its default values. I sometimes use the local provider in a module without declaring a provider block. Let me know if I misunderstand you here though

Reply:
So it sounds like it should allow the situation where TerraformProviders is empty (default) but error on the case where TerraformProviders has a value but the Task's listed provider isn't present.

Sound right?

Reply:
Yes, that sounds right to me! To add onto:

> error on the case where TerraformProviders has a value but the Task's listed provider isn't present

This would be if the TerraformProvider didn't exist any of the listed provider across all tasks. I'm not sure if this validation already exists or if you were planning to add it... I personally don't think that auto-commit requires this validation so ok with it not being included in this PR

Reply:
I was attempting to address the the code comment...

    // TODO: validate providers listed in tasks exist

.. while I was working on validating task/provider inter-dependencies. It isn't directly relevant to this ticket but seemed worth-while as I was almost testing for it already.

So it's not attempting to check that all TerraformProvider entries are used (which seems to be what you're saying in that "This would be.." sentence), but that all the Tasks have valid providers.

Reply:
Gotcha! Just didn't want you to feel obligated. I actually interpret that code comment as you originally had: that all task listed providers should have an associated TerraformProviders block 🤔

Reply:
I reworked this to allow task providers w/o a matching terraform provider as long as the terraform providers is empty. I've added a test to confirm this and adjusted the other test to use a setup with terraform providers set, just not the one the Task wants.

Reply:
the noProvider and noTaskProvider tests look great to me!

For "task providers w/o a matching terraform provider as long as the terraform providers is empty" - I can see a possibility where a module uses an f5 and local provider, where f5 provider needs a TerraformProvider block but the local provider doesn't since it can use default. I think this should be valid but would not pass the validation added here. I'm personally starting to think that there isn't a generalized way to validate the the task provider list to TerraformProvider blocks. Let me know if I misunderstand in anyway.

Reply:
Oh no sorry about my comment I left in. Yeah I had initially imagined that it would be nice to validate and error early. However Lorna is right about the edge case and that it's not necessarily required to have a `terraform_provider` block. It just occurred to me the other day it could also be that they want to use the `driver.terraform { required_providers {} }` block that denotes the namespace.

So removing the added validation (and my troublesome comment) would allow for those two cases.

Reply:
Great. I'll just remove the related code then. Pushing updates soon...

Reply:
I've removed the verify and test code as well as that comment.